code/datums/abilities/critter/mindeater.dm

```diff
@@ -0,0 +1,390 @@
+					if (handholder.item)
+						item_candidates |= handholder.item
+
+		shuffle_list(item_candidates)
+		for (var/i in 1 to min(5, length(item_candidates)))
+			chosen_items += item_candidates[i]
+
+		var/mob/living/critter/mindeater/mindeater = src.holder.owner
+		for (var/obj/item/I as anything in chosen_items)
+			animate(I, 1 SECOND, easing = LINEAR_EASING, alpha = 0)
+			SPAWN(1 SECOND)
+				animate(I, 0.5 SECONDS, alpha = 255)
+				mindeater.levitate_item(I)
+				sleep(10 SECONDS)
+				mindeater.drop_levitated_item(I)
+
+/datum/targetable/critter/mindeater/spatial_swap
+	name = "Spatial Swap"
+	desc = "Swap the location of yourself and another living creature."
+	icon_state = "spatial_swap"
+	cooldown = 20 SECONDS
+	targeted = TRUE
+	target_anything = TRUE
+	max_range = 7
+	pointCost = 25
+
+	tryCast(atom/target)
+		target = src.get_nearest_living(target)
+		if (!target)
+			boutput(src.holder.owner, SPAN_ALERT("You can only target living creatures!"))
+			return CAST_ATTEMPT_FAIL_NO_COOLDOWN
+		return ..()
+
+	cast(atom/target)
+		. = ..()
+		var/mob/living/L = target
+		var/turf/T1 = get_turf(src.holder.owner)
+		var/turf/T2 = get_turf(L)
+		L.set_loc(T1)
+		src.holder.owner.set_loc(T2)
+
+/datum/targetable/critter/mindeater/create
+	name = "Create"
+	desc = "Create a fake Mindeater at the target location."
+	icon_state = "create"
+	cooldown = 45 SECONDS
+	targeted = TRUE
+	target_anything = TRUE
+	pointCost = 15
+
+	cast(atom/target)
+		. = ..()
+		var/obj/dummy/fake_mindeater/fake = new /obj/dummy/fake_mindeater(get_turf(target))
+		fake.set_dir(src.holder.owner.dir)
+
+/datum/targetable/critter/mindeater/disguise
+	name = "Disguise"
+	desc = "Disguise yourself as a creature."
+	icon_state = "disguise"
+	pointCost = 50
+	reveals_on_use = TRUE
+	var/chosen_option
+
+	tryCast(atom/target)
+		src.chosen_option = null
+		var/option = tgui_input_list(src.holder.owner, "What would you like to disguise as?", "Set Disguise", list("Mouse", "Cockroach", "Human"))
+		if (!option)
+			return CAST_ATTEMPT_FAIL_NO_COOLDOWN
+		src.chosen_option = option
+		return ..()
+
+	cast(atom/target)
+		. = ..()
+		var/mob/living/critter/mindeater/mindeater = src.holder.owner
+		mindeater.disguise(src.chosen_option)
+
+		mindeater.abilityHolder.removeAbility(/datum/targetable/critter/mindeater/disguise)
+		mindeater.abilityHolder.addAbility(/datum/targetable/critter/mindeater/clear_disguise)
+
+/datum/targetable/critter/mindeater/clear_disguise
+	name = "Clear Disguise"
+	desc = "Clear your disguise."
+	icon_state = "clear_disguise"
+
+	cast(atom/target)
+		. = ..()
+		var/mob/living/critter/mindeater/mindeater = src.holder.owner
+		mindeater.undisguise()
+
+		mindeater.abilityHolder.removeAbility(/datum/targetable/critter/mindeater/clear_disguise)
+		mindeater.abilityHolder.addAbility(/datum/targetable/critter/mindeater/disguise)
+
+/datum/targetable/critter/mindeater/confuse
+
+/datum/targetable/critter/mindeater/shades
```

Comment:
This is a really cool ability but movement (especially diagonal?) seems to immediately reveal which is the real one because the others don't animate properly.

Reply:
Any suggestions on how to improve? I tried a few things and manually calling Move() seemed to be the best. Could maybe replace /obj/dummy/fake_mindeater with a critter version and have movement_delay() be set to the parent mindeater's movement delay, but maybe better post-initial-playtest?